src/librustc/traits/select.rs

```diff
@@ -2495,6 +2547,40 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
         Ok(VtableFnPointerData { fn_ty: self_ty, nested: obligations })
     }
 
+    fn confirm_generator_candidate(&mut self,
```

Comment:
You mean the 3 functions calling `confirm_poly_trait_refs`?

I would rather see a merge of the 3 locations which deals with trait selection (`select.rs`, `project.rs`, `probe.rs`). Will `Chalk` things help here?